planner/core/planbuilder.go

```diff
@@ -1470,6 +1512,89 @@ func (b *planBuilder) buildExplain(explain *ast.ExplainStmt) (Plan, error) {
 	return p, nil
 }
 
+// colToColumnDef converts expression.Column `c` to ColumnDef:
+// if `c` refers to a column of the source table, we copy some of the column attributes from the referred column.
+// if `c` is an expression(for example, `count(*)`), the original table/column cannot be found.
+func (b *planBuilder) colToColumnDef(c *expression.Column, t *ast.TableName) (*ast.ColumnDef, error) {
+	d := &ast.ColumnDef{}
+	d.Name = &ast.ColumnName{
+		Schema: t.Schema,
+		Table:  t.Name,
+		Name:   c.ColName,
+	}
+	d.Tp = c.GetType().Clone()
+
+	// auto-increment attribute will not be preserved
+	d.Tp.Flag &= ^mysql.AutoIncrementFlag
+	// no index will be preserved
+	d.Tp.Flag &= ^mysql.PriKeyFlag
+	d.Tp.Flag &= ^mysql.UniqueKeyFlag
+	d.Tp.Flag &= ^mysql.MultipleKeyFlag
+
+	table, err := b.is.TableByName(c.DBName, c.OrigTblName)
+	if err != nil {
+		return d, nil
+	}
+	tblInfo := table.Meta()
+
+	isFound := false
+	for _, col := range tblInfo.Columns {
+		if col.Name.L == c.OrigColName.L && tblInfo.Name.L == c.OrigTblName.L {
+			isFound = true
+			// copy column comment
+			if len(col.Comment) > 0 {
+				option := &ast.ColumnOption{
+					Tp:   ast.ColumnOptionComment,
+					Expr: ast.NewValueExpr(col.Comment),
+				}
+				d.Options = append(d.Options, option)
+			}
+
+			// copy column default value
+			switch dv := col.DefaultValue.(type) {
+			case nil:
+				// Do nothing
+			case string:
+				if strings.ToUpper(dv) == strings.ToUpper(ast.CurrentTimestamp) {
+					option := &ast.ColumnOption{
+						Tp:   ast.ColumnOptionDefaultValue,
+						Expr: &ast.FuncCallExpr{FnName: model.NewCIStr(strings.ToUpper(ast.CurrentTimestamp))},
+					}
+					d.Options = append(d.Options, option)
+				} else {
+					option := &ast.ColumnOption{
+						Tp:   ast.ColumnOptionDefaultValue,
+						Expr: ast.NewValueExpr(dv),
+					}
+					d.Options = append(d.Options, option)
+				}
+			case ast.ExprNode:
+				option := &ast.ColumnOption{
+					Tp:   ast.ColumnOptionDefaultValue,
+					Expr: dv,
```

Comment:
Miss Stored field.

Reply:
Yeah but `Stored bool`'s comment says '_Stored is only for ColumnOptionGenerated'_, so in my opinion it's not required here.